packages/roosterjs-editor-plugins/lib/plugins/WatermarkV2/WatermarkV2.ts

```diff
@@ -0,0 +1,119 @@
+import {
+    EditorPlugin,
+    Entity,
+    IEditor,
+    PluginEvent,
+    PluginEventType,
+} from 'roosterjs-editor-types';
+import { kebabCase } from './utils';
+
+const ENTITY_TYPE = 'WATERMARK_WRAPPER';
+const WATERMARK_ATTR = 'data-rooster-watermark';
+const STYLE_ID = 'watermarkStyle';
+
+const DEFAULT_STYLE = {
+    color: 'rgba(0,0,0,0.3)',
+    position: 'absolute',
+    cursor: 'text',
+};
+
+/**
+ * Yet another Watermark plugin
+ */
+export default class WatermarkV2 implements EditorPlugin {
+    private editor: IEditor;
+
+    /**
+     * Create an instance of Watermark plugin
+     * @param watermark The watermark string
+     * @param style watermark style
+     */
+    constructor(private watermark: string, private readonly style?: CSSStyleDeclaration) {
+        this.style = {
+            ...DEFAULT_STYLE,
+            ...this.style,
+            content: `attr(${WATERMARK_ATTR})`, // key trick
+        };
+    }
+
+    /**
+     * Get a friendly name of this plugin
+     */
+    getName() {
+        return 'WatermarkV2';
+    }
+
+    /**
+     * Initialize this plugin. This should only be called from Editor
+     * @param editor Editor instance
+     */
+    initialize(editor: IEditor) {
+        this.editor = editor;
+
+        const doc = this.editor.getDocument();
+        let styleElement = doc.getElementById(STYLE_ID) as HTMLStyleElement;
+        if (!styleElement) {
+            styleElement = doc.createElement('style');
+            doc.head.appendChild(styleElement);
+            styleElement.id = STYLE_ID;
+        }
+        const cssText = Object.keys(this.style)
+            .map(key => `${kebabCase(key)}:${this.style[key as any]}`)
+            .join(';');
+        const className = this.editor.getEditorDomAttribute('class').split(' ')[0];
```

Comment:
If editor DIV is using a shared class name with other elements, this may have some issue. 

We can leverage the existing unique ID strategy that we are using for table selection (and the code to manage style tags as well) to make this to be more robust.